src/ocf_core.c

```diff
@@ -142,11 +142,11 @@ int ocf_core_visit(ocf_cache_t cache, ocf_core_visitor_t visitor, void *cntx,
 
 /* *** HELPER FUNCTIONS *** */
 
-static uint32_t _calc_dirty_for(uint64_t dirty_since)
+static uint64_t _calc_dirty_for(uint64_t dirty_since)
 {
-	return dirty_since ?
-		(env_ticks_to_msecs(env_get_tick_count() - dirty_since) / 1000)
-		: 0;
+	uint64_t current_time = env_ticks_to_secs(env_get_tick_count());
+
+	return dirty_since ? ((current_time - dirty_since) % ULONG_MAX) : 0;
```

Comment:
how does this compare to env_time_after?

Reply:
`env_time_after` takes 2 agruments and returns boolean value - is parameter A value greater than parameter B value? Not applicable here.

Reply:
ok, thanks